pkg/query-service/app/querier/helper.go

```diff
@@ -208,14 +292,30 @@ func (q *querier) runBuilderQuery(
 		Name:   queryName,
 		Series: mergedSeries,
 	}
-	// Cache the seriesList for future queries
-	if len(missedSeries) > 0 && !params.NoCache && q.cache != nil {
-		mergedSeriesData, err := json.Marshal(mergedSeries)
+
+	var mergedSeriesData []byte
+	missedSeriesLen := len(missedSeries)
+	if missedSeriesLen > 0 && !params.NoCache && q.cache != nil {
+		// caching the data
+		var err error
+		mergedSeriesData, err = json.Marshal(mergedSeries)
 		if err != nil {
 			zap.S().Error("error marshalling merged series", zap.Error(err))
 			return
```

Comment:
Same, don't return if there are errors related to cache stuff. If you see other parts of the such as this https://github.com/SigNoz/signoz/blob/ab4f6adb198363c4efa7bf00d67342b80f38d74f/pkg/query-service/app/querier/helper.go#L282-L284, it doesn't return prematurely. The context when return is used is important.

Reply:
make sense, thanks